topics/dev/tutorials/tool-from-scratch/tutorial.md

```diff
@@ -0,0 +1,1182 @@
+  - Understand how to write a Galaxy tool wrapper
+  - Be able to use a tool wrapper locally
+  - Be able to test the tool wrapper
+  - Learn how to add tools to the Galaxy toolshed and make them available to any user
+time_estimation: 3H
+key_points:
+- Galaxy tools are made in 3 parts -- Conda, tool wrappers, and planemo
+  - Anyone can write and deploy a tool wrapper!
+contributors:
+  - astrovsky01
+  - davebx
+  - bernt-matthias
+
+---
+
+
+# Introduction
+{:.no_toc}
+
+Tools wrappers allow any command line runnable code or programs to be run inside a galaxy environment.
+Although Galaxy has thousands of tools readily available, new software and packages will always be useful.
+This tutorial is designed to allow anyone to create, run, and deploy new tools in a Galaxy environment.
+
+This tutorial is intended to instruct the reader in the full process of integrating a tool into Galaxy.
+In three parts, it will cover the creation of a bioconda recipe for a new tool, writing a Galaxy tool
+wrapper, and finally the testing and deployment of this tool into both a local and public Galaxy environment.
+Although this will include specific tools, the training is generalizable.
+
+> ### Agenda
+>
+> In this tutorial, we will cover:
+>
+> 1. TOC
+> {:toc}
+>
+{: .agenda}
+
+# Galaxy Tools and Conda
+
+Galaxy tools run within a conda environment, and are able to run any tool capable of running inside
+that type of environment. The wrapper includes a set of conda packages available from either bioconda
+or conda forge, and retrieves them on tool installation. Sometimes this may be a single package, as
+with many command line tools, other times, there may be several. The first part of this tutorial discusses
+the creation and deployment of these packages, making them available for Galaxy to retrieve.
+
+## Writing a Bioconda Recipe
+
+Bioconda is a repository of conda packages for software and tools relevant to the life sciences. Using conda
+packages ensures better reproducibility, since each conda package is usually immutable, an exception being
+if it is later discovered that the software in the package had always been non-functional and should be replaced.
+Recipes should always define the following 6 sections in the `meta.yaml` file:
+
+- package
+- source
+- build
+- requirements
+- test
+- about
+
+> ### {% icon hands_on %} Hands-on: Writing a Bioconda Recipe
+>
+> 1. The first thing we should do is prepare our workspace. We create a branch in git, a conda environment with pip and
+> conda-build preinstalled, and a folder for the meta.yaml file.
+>
+>    > ### {% icon code-in %} Input: Bash
+>    > ```bash
+>    > git checkout -b bellerophon_bioconda
+>    > conda create -y --name bellerophon_bioconda pip python conda-build
+>    > conda activate bellerophon_bioconda
+>    > mkdir recipes/bellerophon
+>    > ```
+>    {: .code-in}
+>
+> 2. Next we determine the SHA-256 checksum of the source tarball.
+>
+>    > ### {% icon code-in %} Input: Bash
```

Comment:
if you have a numbered section with some pre blocks, you need to indent those to match up with the text after the number (so 3 spaces for single digits), rather than being left aligned.

Reply:
I'm not sure what you mean? When I try that, it no longer numbers properly, so I think I'm doing it wrong

Reply:
![image](https://user-images.githubusercontent.com/458683/121941814-04690000-cd50-11eb-965b-0a18922c5b39.png)

when it's indented like that, that generally triggers re-numbering at least in the GTN's markdown processor, it sees each item as a new list and restarts at 1.